core/src/tangram.h

```diff
@@ -184,29 +181,8 @@ class Map {
 
 private:
 
-    enum class EaseField { position, zoom, rotation, tilt };
-
-    void setEase(EaseField _f, Ease _e);
-    void clearEase(EaseField _f);
-    void setScene(std::shared_ptr<Scene>& _scene);
-    void setPositionNow(double _lon, double _lat);
-    void setZoomNow(float _z);
-    void setRotationNow(float _radians);
-    void setTiltNow(float _radians);
-
-    std::mutex m_tilesMutex;
-    std::mutex m_sceneMutex;
-    std::unique_ptr<TileWorker> m_tileWorker;
-    std::unique_ptr<TileManager> m_tileManager;
-    std::shared_ptr<Scene> m_scene;
-    std::shared_ptr<Scene> m_nextScene;
-    std::shared_ptr<View> m_view;
-    std::unique_ptr<Labels> m_labels;
-    std::unique_ptr<InputHandler> m_inputHandler;
-    std::unique_ptr<AsyncWorker> m_asyncWorker;
-    std::vector<SceneUpdate> m_sceneUpdates;
-    std::array<Ease, 4> m_eases;
-    bool m_cacheGlState;
+    class Impl;
+    std::unique_ptr<Impl> impl;
```

Comment:
`Map` is implicitly non-copyable - the compiler can't generate a copy constructor because of the `unique_ptr`

Supporting move semantics for `Map` would be trivial, I don't know if there's a serious need for it though.